kw6/stream.py

```diff
@@ -45,7 +45,37 @@ def from_path(path, header_path=None):
             cached_positions=header_positions(header_path) if header_path is not None else {},
         )
 
-    def seek_(self, frame_index: types.FRAME_INDEX):
+    def __getitem__(self, indices_or_slice):
+        if type(indices_or_slice) == int:
+            if indices_or_slice < 0:
+                raise IndexError('Negative indexing not supported.')
+            self._seek(indices_or_slice)
+            if not self.empty():
+                positions = Position.from_stream_(self.stream)
+            else:
+                raise IndexError(indices_or_slice)
+
+        elif type(indices_or_slice) == slice:
+            if indices_or_slice.start is None:
+                return self[0: indices_or_slice.stop]
+            if indices_or_slice.stop is None:
+                self._seek(indices_or_slice.start)
+                positions = [position for position in self]
+            else:
+                positions = [
+                    self[index]
+                    for index in range(indices_or_slice.start, indices_or_slice.stop)
+                ]
+
+        elif type(indices_or_slice) == list or type(indices_or_slice) == tuple:
+            positions = [self[index] for index in indices_or_slice]
+
+        return positions
+
+    def empty(self):
+        return self.stream.peek(1) == b''
+
+    def _seek(self, frame_index: types.FRAME_INDEX):
```

Comment:
Can this be a public method? Would be nice to end the name in a `_` and I guess you didn't want to because it was ugly to call it `_seek_`?

Reply:
That was exactly the thought process :smile: But sure, it could actually make sense to have it be a public method since you might want to call `seek_(0)` to make `__iter__` run from the start.

Reply:
Done